src/main.cpp

```diff
@@ -2080,6 +2081,47 @@ void PartitionCheck(bool (*initialDownloadCheck)(), CCriticalSection& cs, const
     }
 }
 
+Consensus::VersionBitsConditionChecker checkCSV(Consensus::DEPLOYMENT_CSV);
+Consensus::VersionBitsConditionChecker* const softforkcheck[] = {&checkCSV};
+
+int32_t ComputeBlockVersion(const CBlockIndex* pindexPrev, const Consensus::Params& params)
+{
+    LOCK(cs_main);
+    int32_t nVersion = 0;
+
+    for (unsigned int i = 0; i < sizeof(softforkcheck) / sizeof(softforkcheck[0]); i++) {
+        if (softforkcheck[i]->IsSetFor(pindexPrev, params)) {
+            nVersion |= softforkcheck[i]->Mask(params);
+        }
+    }
+
+    if (nVersion == 0) {
```

Comment:
Assume the start date for the first bip9 softfork is april 1st, but we
release code with bip9 and that softfork on march 1st.

That means that it's likely that during march no new-style nVersion will be
mined. That also means that anyone who does run upgraded code will get
warnings, because the nVersion they expect (0x20000000) differs from the
one they see (4).